cmd/gitopsUpdateDeployment_test.go

```diff
@@ -0,0 +1,260 @@
+package cmd
+
+import (
+	"errors"
+	gitUtil "github.com/SAP/jenkins-library/pkg/git"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestErrorOnTempDir(t *testing.T) {
+	defer func() {
+		fileUtilities = piperutils.Files{}
+	}()
+
+	fileUtilities = filesMockErrorTempDirCreation{}
+
+	var c gitopsExecRunner
+	var configuration = &gitopsUpdateDeploymentOptions{
+		BranchName:           "main",
+		CommitMessage:        "This is the commit message",
+		ServerURL:            "https://github.com",
+		Username:             "admin3",
+		Password:             "validAccessToken",
+		FilePath:             "dir1/dir2/depl.yaml",
+		ContainerName:        "myContainer",
+		ContainerRegistryURL: "https://myregistry.com/registry/containers",
+		ContainerImage:       "myFancyContainer:1337",
+	}
+
+	err := runGitopsUpdateDeployment(configuration, c)
```

Comment:
Yes, I also stumbled about this when I refactored the tests for the git.go file. Handle them as parameter rather than global variables is more secure and can be parallelized in the tests.
Will do so.

Reply:
done